src/aws_encryption_sdk/keyring/raw_keyring.py

```diff
@@ -0,0 +1,221 @@
+                                    required to encrypt data key
+        :type encryption_materials: aws_encryption_sdk.materials_managers.EncryptionMaterials
+        :returns: Contains signing key, unencrypted data key, encrypted data keys,
+                                    encryption context and algorithm suite required to encrypt data key
+        :rtype: aws_encryption_sdk.materials_managers.EncryptionMaterials
+        """
+
+        encryption_materials = on_encrypt_helper(
+            encryption_materials, self._key_provider, self._wrapping_key, self._wrapping_algorithm, self.key_name
+        )
+
+        return encryption_materials
+
+    def on_decrypt(self, decryption_materials, encrypted_data_keys):
+        """Attempt to decrypt the encrypted data keys.
+
+        :param decryption_materials: Contains verification key, algorithm, encryption context and keyring trace.
+        :type decryption_materials: aws_encryption_sdk.materials_managers.DecryptionMaterials
+        :param encrypted_data_keys: List of encrypted data keys.
+        :type: List of `aws_encryption_sdk.structures.EncryptedDataKey`
+        :returns: Contains verification key, algorithm, data_encryption_key, encryption context and keyring trac
+        :rtype: aws_encryption_sdk.materials_managers.DecryptionMaterials
+        """
+
+        # Decrypt data key
+        expected_key_info_len = len(self._key_info_prefix) + self._wrapping_algorithm.algorithm.iv_len
+        if (
+            encrypted_data_keys.key_provider.provider_id == self._key_provider.provider_id
+            and len(encrypted_data_keys.key_provider.key_info) == expected_key_info_len
+            and encrypted_data_keys.key_provider.key_info.startswith(self._key_info_prefix)
+        ):
+            decryption_materials = on_decrypt_helper(
+                decryption_materials, self._wrapping_key, self._wrapping_algorithm, self.key_name, encrypted_data_keys
+            )
+
+        return decryption_materials
+
+
+class RawRSAKeyring(Keyring):
+    """Public class for Raw RSA Keyring.
+
+    :param str key_namespace: String defining the keyring ID
+    :param bytes key_name: Key ID
+    :param wrapping_key: Encryption key with which to wrap plaintext data key
+    :type wrapping_key: WrappingKey
+    :param wrapping_algorithm: Wrapping Algorithm with which to wrap plaintext data key
+    :type wrapping_algorithm: WrappingAlgorithm
+    :param key_provider: Complete information about the key in the keyring
+    :type key_provider: MasterKeyInfo
+    """
+
+    key_namespace = attr.ib(validator=attr.validators.instance_of(str))
+    key_name = attr.ib(hash=True, validator=attr.validators.instance_of(bytes))
+    _wrapping_key = attr.ib(hash=True, repr=False, validator=attr.validators.instance_of(WrappingKey))
+    _wrapping_algorithm = attr.ib(repr=False, validator=attr.validators.instance_of(WrappingAlgorithm))
+
+    _key_provider = MasterKeyInfo(provider_id=key_namespace, key_info=key_name)
```

Comment:
This needs to go in a `__attrs_post_init__` method.

Reply:
Made all changes